components/dash-core-components/src/fragments/Dropdown.react.js

```diff
@@ -47,12 +47,26 @@ const Dropdown = props => {
     } = props;
     const [optionsCheck, setOptionsCheck] = useState(null);
     const [sanitizedOptions, filterOptions] = useMemo(() => {
-        const sanitized = sanitizeOptions(options);
+        let sanitized = sanitizeOptions(options);
+        const firstOption = sanitized ? head(sanitized) : null;
+        let labelKey = 'label';
+        if (firstOption && firstOption.search) {
+            labelKey = 'search';
+        } else if (firstOption && React.isValidElement(firstOption.label)) {
```

Comment:
I can see people using strings for some options and components for others - like if they want a few options to have custom font color but others just use the default behavior. So just looking at the first one may not be robust. Also wondering if there's a way to pull the text out of a component for the same use case, particularly when the value is encoded in a way that isn't useful for searching (matching names to ID numbers or something). But it could still be nice to search on value too, especially if the search would match EITHER label or value. Is that possible?

Reply:
> But it could still be nice to search on value too, especially if the search would match EITHER label or value. Is that possible?

It's possible to add field indexes to the search, maybe we should add a new prop `search_indexes`, then it could match any field in a dictionary of options ?